test/common/stats/BUILD

```diff
@@ -99,6 +99,18 @@ envoy_cc_test(
     ],
 )
 
+envoy_cc_test_library(
```

Comment:
nice - thank you!

Reply:
can you pull this out and make a separate small PR for this test infra refactor?

Reply:
ping on this -- I'd be happy to review this and other small parts of this PR separately, first.

Reply:
PTAL https://github.com/envoyproxy/envoy/pull/24887